### PR TITLE
#159965311 Ft follow and unfollow a user 

### PR DESCRIPTION
**Pull request by mikenthiwa**
#### What does this PR do?
- Allows users to follow and unfollow each other

#### Description of task to be completed?
 - A User can choose who to follow and unfollow.
 - A user can also view all users who are following him\her and who he\she is following.

#### How should this be manually tested?
$ git clone https://github.com/andela/ah-magnificent6.git
$ git checkout ft-follow-unfollow-user-#159965311
$ cd ah-magnificent6
$ cp .env-sample .env [and set the environment variables]
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip install -r requirements.txt
(venv) $ python manage.py migrate
(venv) $ python manage.py runserver

Using [Postman](https://www.getpostman.com/) navigate to 
User registration: `POST {your-server-root}:8000/api/user/signup`
```Register and login two users to effeciently test the feature```

```
{
    "email": "user1@user.user",
    "username": "user1",
    "password": "User$#@!"
}
```

User Login: `POST {your-server-root}:8000/api/user/login`
```
{
    "email": "user1@user.user",
    "password": "User$#@!"
}
Repeat the same process for user2
```
Use the token from the response in the login as the authentication header while following, unfollowing, get all following and followed.

**Follow a user**:
NB: http request method: `POST`
 - You have to provide the username of the user you want to follow.
 - For instance i have logged in as user1:
 - In Postman client: `POST {your-server-root}:8000/api/profile/user2/follow`
 - Provide token. Bearer @#$%^123456789QWERTYU 
 - Returns your profile
screen shot 2018-09-13 at 11 47 42

Un-follow a user:
NB: http request method: `DELETE`
 - You have to provide the username of the user you want to follow.
 - In Postman client: `DELETE {your-server-root}:8000/api/profile/user2/follow`
 - Provide token. Bearer @#$%^123456789QWERTYU 
 - Returns your profile
screen shot 2018-09-13 at 12 15 02

Users following you:
NB: http request method: `GET`
 - You have to provide your username.
 - In Postman client: `GET {your-server-root}:8000/api/profile/user1/followers`
 - Provide token. Bearer @#$%^123456789QWERTYU 
 - Returns the profiles of your followers
screen shot 2018-09-13 at 12 23 37

Users you are following:
NB: http request method: `GET`
 - You have to provide your username.
 - In Postman client: `GET {your-server-root}:8000/api/profile/user1/following`
 - Provide token. Bearer @#$%^123456789QWERTYU 
 - Returns the profiles of users following you
screen shot 2018-09-13 at 12 30 52

### CODE REVIEW

**Comment by meshack-mbuvi** [comment]
We agreed that we should not lower test coverage. We either maintain it at where it is or increase it. I, therefore, request that you fix that.

**Comment by micahoriaso** [comment]
Please ignore the node_modules directory

File: authors/apps/profiles/models.py [hunk 1]
```diff
@@ -18,29 +19,49 @@ class Profile(TimeStampModel):
     country = models.CharField(_('country'), blank=True, null=True, max_length=100, default='')
     phone = models.IntegerField(_('phone'), blank=True, null=True, default=0)
     website = models.URLField(_('website'), blank=True, null=True, default='')
-    # follows = models.ManyToManyField('self', related_name='followed_by', symmetrical=False)
-    
+    follows = models.ManyToManyField('self', related_name='followed_by', symmetrical=False)
+    username = models.CharField(_('username'), max_length=100, blank=True, null=True)
```
Comment (pngg) [inline authors/apps/profiles/models.py]: Could we refactor this `username` to use a `SerializerMethodField` or `ReadOnlyField` in `ProfileSerializer` instead?

File: authors/urls.py [hunk 2]
```diff
@@ -28,4 +28,5 @@
                          namespace='authentication')),
     path('api/', include('authors.apps.articles.urls', namespace='articles')),
     path('', schema_view),
+    path('api/', include('authors.apps.profiles.urls', namespace='profiles')),
```
Comment (parseen254) [inline authors/urls.py]: @mikenthiwa lets go with `api/profiles` here.

**Review by parseen254 — CHANGES_REQUESTED** [review]
And remove the `node_modules` directory from our git history.

File: authors/urls.py (hunk 2 above)
Comment (mikenthiwa) [reply authors/urls.py]: Done

File: authors/apps/profiles/models.py (hunk 1 above)
Comment (mikenthiwa) [reply authors/apps/profiles/models.py]: Done

File: authors/urls.py (hunk 2 above)
Comment (mikenthiwa) [reply authors/urls.py]: Done

**Pull request merged**